sql/src/main/java/io/crate/analyze/expressions/ExpressionAnalyzer.java

```diff
@@ -448,13 +449,15 @@ protected Symbol visitExpression(Expression node, ExpressionAnalysisContext cont
 
         @Override
         protected Symbol visitCurrentTime(CurrentTime node, ExpressionAnalysisContext context) {
-            if (!node.getType().equals(CurrentTime.Type.TIMESTAMP)) {
+            boolean isTime = node.getType().equals(CurrentTime.Type.TIME);
+            if (!(node.getType().equals(CurrentTime.Type.TIMESTAMP) || isTime)) {
                 visitExpression(node, context);
             }
+            String functionName = isTime ? CurrentTimeFunction.NAME : CurrentTimestampFunction.NAME;
```

Comment:
The conditions seem a bit redundant - maybe this could be changed to avoid the duplicate `isTime` checks - I think then it would also be easier to read.

Reply:
I have removed the implementation of CurrentTimeFunction from this PR.
It was dragged in by mistake, as I rebased from the branch I use for it.
The plan is to get TimeType in and then use it to move forward with the function. I'd rather not mix.